pkg/cloudprovider/aws/suite_test.go

```diff
@@ -165,16 +173,111 @@ var _ = Describe("Allocation", func() {
+					node := ExpectScheduled(ctx, env.Client, pod)
+					Expect(node.Labels).To(HaveKeyWithValue(InstanceTypeLabel, "inf1.2xlarge"))
+					nodeNames.Insert(node.Name)
+				}
+				Expect(nodeNames.Len()).To(Equal(2))
+			})
+			It("should launch instances of different type on second recon attempt with Insufficient Capacity Error Cache fallback", func() {
+				fakeEC2API.InsufficientCapacityPools = []fake.CapacityPool{{InstanceType: "p3.8xlarge", Zone: "test-zone-1a"}}
+				pods := ExpectProvisioned(ctx, env.Client, scheduler, provisioners, provisioner,
+					test.UnschedulablePod(test.PodOptions{
+						ResourceRequirements: v1.ResourceRequirements{
+							Requests: v1.ResourceList{resources.NvidiaGPU: resource.MustParse("1")},
+							Limits:   v1.ResourceList{resources.NvidiaGPU: resource.MustParse("1")},
+						},
+					}),
+					// Should pack onto same instance
+					test.UnschedulablePod(test.PodOptions{
+						ResourceRequirements: v1.ResourceRequirements{
+							Requests: v1.ResourceList{resources.NvidiaGPU: resource.MustParse("2")},
+							Limits:   v1.ResourceList{resources.NvidiaGPU: resource.MustParse("2")},
+						},
+					}),
+					// Should pack onto a separate instance
+					test.UnschedulablePod(test.PodOptions{
+						ResourceRequirements: v1.ResourceRequirements{
+							Requests: v1.ResourceList{resources.NvidiaGPU: resource.MustParse("4")},
+							Limits:   v1.ResourceList{resources.NvidiaGPU: resource.MustParse("4")},
+						},
+					}),
+				)
+				// it should've tried to pack them in test-zone-1a on p3.8xlages then hit ICEs, so the next attempt will avoid that pool and try test-zone-1b
+				for _, pod := range pods {
+					ExpectNotScheduled(ctx, env.Client, pod)
 				}
-				Expect(InstancesLaunchedFrom(fakeEC2API.CalledWithCreateFleetInput.Iter())).To(Equal(2))
-				overrides := []*ec2.FleetLaunchTemplateOverridesRequest{}
-				for input := range fakeEC2API.CalledWithCreateFleetInput.Iter() {
-					overrides = append(overrides, input.(*ec2.CreateFleetInput).LaunchTemplateConfigs[0].Overrides...)
+
+				nodeNames := sets.NewString()
+				for _, pod := range ExpectProvisioned(ctx, env.Client, scheduler, provisioners, provisioner, pods...) {
+					node := ExpectScheduled(ctx, env.Client, pod)
+					Expect(node.Labels).To(SatisfyAll(
+						HaveKeyWithValue(InstanceTypeLabel, "p3.8xlarge"),
+						HaveKeyWithValue(ZoneLabel, "test-zone-1b")))
+					nodeNames.Insert(node.Name)
+				}
+				Expect(nodeNames.Len()).To(Equal(2))
+			})
+			It("should launch instances on later recon attempt with Insufficient Capacity Error Cache expiry", func() {
+				fakeEC2API.InsufficientCapacityPools = []fake.CapacityPool{{InstanceType: "inf1.6xlarge", Zone: "test-zone-1a"}}
+				pods := ExpectProvisioned(ctx, env.Client, scheduler, provisioners, provisioner,
+					test.UnschedulablePod(test.PodOptions{
+						ResourceRequirements: v1.ResourceRequirements{
+							Requests: v1.ResourceList{resources.AWSNeuron: resource.MustParse("2")},
```

Comment:
It might be cleaner to do 
```
test.PodOptions{
  NodeSelector: map[string]string{v1.LabelInstanceTypeStable: "inf1.6xlarge"}
}
```
And explicitly test an iced instance type. I worry that if someone adds a new instance type for testing, then this test will break. I might also simplify to a single pod instead of two. In general, test testing style in karpenter is to be surgical and focus on the specific behavior being tested, but also exercise the e2e happy code path outside of the specific behavior..